src/scripts/main.js

```diff
@@ -1,3 +1,317 @@
+      10,
+    );
+    const salaryB = parseInt(
+      rowB
+        .querySelector('td:nth-child(5)')
+        .textContent.trim()
+        .replace(/[$,]/g, ''),
+      10,
+    );
+
+    return isAscending ? salaryA - salaryB : salaryB - salaryA;
+  });
+
+  tbody.innerHTML = '';
+
+  sortedRows.forEach((row) => tbody.appendChild(row));
+
+  isAscending = !isAscending;
+});
+
+// active row
+rows.forEach((row) => {
+  row.addEventListener('click', () => {
+    rows.forEach((r) => r.classList.remove('active'));
+
+    row.classList.add('active');
+  });
+});
+
+// create form
+const form = document.createElement('form');
+
+form.classList.add('new-employee-form');
+
+const inputs = [
+  {
+    label: 'Name:',
+    name: 'name',
+    type: 'text',
+    qa: 'name',
+  },
+
+  {
+    label: 'Position:',
+    name: 'position',
+    type: 'text',
+    qa: 'position',
+  },
+
+  {
+    label: 'Age:',
+    name: 'age',
+    type: 'number',
+    qa: 'age',
+  },
+
+  {
+    label: 'Salary',
+    name: 'salary',
+    type: 'number',
+    qa: 'salary',
+  },
+];
+
+inputs.forEach((input) => {
+  const label = document.createElement('label');
+
+  label.textContent = input.label;
+
+  const inputElement = document.createElement('input');
+
+  inputElement.name = input.name;
+  inputElement.type = input.type;
+  inputElement.setAttribute('data-qa', input.qa);
+
+  label.appendChild(inputElement);
+  form.appendChild(label);
+  form.appendChild(document.createElement('br'));
+});
+
+const officeLabel = document.createElement('label');
+
+officeLabel.textContent = 'Office';
+
+const select = document.createElement('select');
+
+select.name = 'office';
+select.setAttribute('data-qa', 'office');
+
+[
+  'Tokyo',
+  'Singapore',
+  'London',
+  'New York',
+  'Edinburgh',
+  'San Francisco',
+].forEach((city) => {
+  const option = document.createElement('option');
+
+  option.value = city;
+  option.textContent = city;
+  select.appendChild(option);
+});
+
+officeLabel.appendChild(select);
+form.appendChild(officeLabel);
+form.appendChild(document.createElement('br'));
+
+document.body.appendChild(form);
+
+const submitButton = document.createElement('button');
+
+submitButton.type = 'submit';
+submitButton.textContent = 'Save to table';
+
+form.appendChild(submitButton);
+
+form.addEventListener('submit', function (e) {
+  e.preventDefault();
+
+  const formName = form.querySelector('input[name="name"]').value.trim();
+  const formPosition = form
+    .querySelector('input[name="position"]')
+    .value.trim();
+  const formAge = parseInt(
+    form.querySelector('input[name="age"]').value.trim(),
+    10,
+  );
+  const formSalary = parseInt(
+    form
+      .querySelector('input[name="salary"]')
+      .value.trim()
+      .replace(/[$,]/g, ''),
+    10,
+  );
+
+  const formOffice = form.querySelector('select[name="office"]').value;
+
+  if (
+    !formName ||
+    !formOffice ||
+    !formPosition ||
+    isNaN(formAge) ||
+    isNaN(formSalary)
```

Comment:
The form validation logic does not check if the salary input is a valid number before parsing it. Consider adding a check to ensure that the salary input is a valid number to prevent potential errors.